pkg/serviceaccounts/controllers/create_dockercfg_secrets.go

```diff
@@ -317,45 +380,12 @@ func (e *DockercfgController) createTokenSecret(serviceAccount *api.ServiceAccou
 		Type: api.SecretTypeServiceAccountToken,
 		Data: map[string][]byte{},
 	}
-
-	_, err := e.client.Secrets(tokenSecret.Namespace).Create(tokenSecret)
-	if err != nil {
-		return nil, err
-	}
-
-	// now we have to wait for the service account token controller to make this valid
-	// TODO remove this once we have a create-token endpoint
-	for i := 0; i <= tokenSecretWaitTimes; i++ {
-		liveTokenSecret, err2 := e.client.Secrets(tokenSecret.Namespace).Get(tokenSecret.Name)
-		if err2 != nil {
-			return nil, err2
-		}
-
-		if len(liveTokenSecret.Data[api.ServiceAccountTokenKey]) > 0 {
-			return liveTokenSecret, nil
-		}
-
-		time.Sleep(wait.Jitter(tokenSecretWaitInterval, 0.0))
-
-	}
-
-	// the token wasn't ever created, attempt deletion
-	glog.Warningf("Deleting unfilled token secret %s/%s", tokenSecret.Namespace, tokenSecret.Name)
-	if deleteErr := e.client.Secrets(tokenSecret.Namespace).Delete(tokenSecret.Name); (deleteErr != nil) && !kapierrors.IsNotFound(deleteErr) {
-		utilruntime.HandleError(deleteErr)
-	}
-	return nil, fmt.Errorf("token never generated for %s", tokenSecret.Name)
+	return e.client.Secrets(tokenSecret.Namespace).Create(tokenSecret)
 }
 
 // createDockerPullSecret creates a dockercfg secret based on the token secret
-func (e *DockercfgController) createDockerPullSecret(serviceAccount *api.ServiceAccount) (*api.Secret, error) {
+func (e *DockercfgController) createDockerPullSecret(tokenSecret *api.Secret, serviceAccount *api.ServiceAccount) error {
```

Comment:
and the only changes to `createDockerPullSecret` would be to short-circuit if `createTokenSecret` says it didn't create a populated secret:
```
tokenSecret, isPopulated, err := e.createTokenSecret(serviceAccount)
if err != nil {
  return nil, false, err
}
if !isPopulated {
  return nil, false, nil
}
```

and to only return `true` for created if it makes it to the end of the function